distributed/client.py

```diff
@@ -2571,6 +2571,7 @@ def __anext__(self):
             raise StopAsyncIteration  # flake8: noqa
         while self.queue.empty():
             yield self.event.wait()
+        sleep(0.0000000001)
```

Comment:
For the record, there are subtle differences between `yield`-based coroutines and `await`-based coroutines (the former will always yield to the scheduler while the second might not, which can create deadlocks). I'm not really an expert on `await`. I also don't know whether it's ok to make `__await__` a `yield`-based coroutine. Perhaps @bdarnell has more insight on this.